src/appmixer/utils/controls/OnError/OnError.js

```diff
@@ -1,59 +1,94 @@
+        } catch (error) {
+            console.error('Error fetching errors:', error);
+            await context.log({ level: 'error', error: error.message || error });
+            break;
+        }
     }
 
-    return getErrors(context, now, errors, result.nextFrom, size);
+    return errors;
 }
 
-/**
- * Add Flow name and component labels to those errors.
- * @param {Context} context
- * @param {Array} errors
- */
+// Adds labels to errors for better identification
 function addLabels(context, errors) {
 
-    return errors.map(err => {
-        return Object.assign(
-            err,
-            {
-                flowName: context.flowInfo.flowName
-            },
-            err.componentId ? {
-                componentLabel: context.flowDescriptor[err.componentId].label || err.componentType.split('.')[3]
-            } : {});
-    });
+    return errors.map(err => ({
+        ...err,
+        flowName: context.flowInfo.flowName,
+        componentLabel: err.componentId ? (context.flowDescriptor[err.componentId]?.label || err.componentType.split('.')[3]) : undefined
+    }));
 }
 
 module.exports = {
 
-    async start(context) {
+    async tick(context) {
+
+        const lockName = `errorProcessingLock-${context.componentId}`; // Define a unique lock name
+        let lock;
+        try {
+            lock = await context.lock(lockName, { ttl: context.config.lockTTL || 1000 * 60 * 5, maxRetryCount: 0 });
+            const { lastLogId, gridTimestamp } = context.state;
+            // ensure that if there is a limit set by context.config.limit
+            const limit = context.properties.limit
+                ? Math.min(
+                    parseInt(context.properties.limit),
+                    parseInt(context.config.limit) || Infinity
+                )
+                : parseInt(context.config.limit) || 100;
 
-        return context.saveState({ since: new Date().toISOString() });
-    },
+            const result = await fetchErrorsPage(context, lock, limit, lastLogId, gridTimestamp);
 
-    async tick(context) {
+            if (result?.length > 0) {
+                // Process and send the errors to outport
+                const labeledErrors = addLabels(context, result);
+                await context.sendArray(labeledErrors, 'out');
 
-        const now = new Date().toISOString();
-        const errors = await getErrors(context, now);
-        await context.sendArray(addLabels(context, errors), 'out');
-        return context.saveState({ since: now });
+                // Update state with the _id of the last log in the page
+                const newLastLogId = result[result.length - 1]['_id'];
+                const newGridTimestamp = result[result.length - 1]['gridTimestamp'];
+                return context.saveState({ lastLogId: newLastLogId, gridTimestamp: newGridTimestamp });
+            }
+        } catch (error) {
+            await context.log({ level: 'error', message: error.message || error });
```

Comment:
There is no need for this construct. If you let the error propagate, the system will log it.